app/tests/test_build_browse_consignment_query.py

```diff
@@ -66,3 +70,32 @@ def test_build_browse_consignment_query_no_results(
         )
         results = query.all()
         assert results == []
+
+    def test_build_browse_consignment_query_filters_by_end_date_when_available(
+        self, client: FlaskClient, mock_standard_user, browse_consignment_files
+    ):
+        """
+        Given a consignment with files that have both end_date and date_last_modified
+        When build_browse_consignment_query is called with date_last_modified filtering
```

Comment:
isn't it more appropriate for the filed to be Date of record which then uses either date_last_modified or end_date behind the scenes?